server/server.cc

```diff
@@ -41,15 +45,51 @@ void Session::do_read() {
     [this, self](boost::system::error_code ec, std::size_t len) {
       if (!ec) {
         printf("Incoming Data length %lu:\n", len);
-        do_write();
+        std::unique_ptr<Request> req = Request::Parse(data_);
+        Response* response = new Response;
+        std::string response_string;
+        if(req.get()) {
+          // response valid 
+          printf("request key: %s\n", req->uri().c_str());
+          std::string key = "";
+          for(const auto & key_match : *handler_->RequestHandlers) {
+            auto res = std::mismatch(key_match.first.begin(), key_match.first.end(), req->uri().begin());
+            if(res.first == key_match.first.end()) { //match found
+              std::string tmp_key = std::string(key_match.first.begin(), res.first);
+              if(tmp_key.size() > key.size()) { //if the match is better, that's our new key'
+                
+                key = tmp_key;
```

Comment:
I think for here it may be better to change the "key" to "max_key" or add some comments so that we know it is trying to find the longest matching here. Also it may be helpful to comment about what exactly key_match has here so that readers don't need to go back and forth between files to figure out what's happening